modules/sdk/src/main/scala/org/tessellation/sdk/domain/snapshot/services/GlobalL0Service.scala

```diff
@@ -37,20 +50,36 @@ object GlobalL0Service {
     l0GlobalSnapshotClient: L0GlobalSnapshotClient[F],
     globalL0ClusterStorage: L0ClusterStorage[F],
     lastGlobalSnapshotStorage: LastSnapshotStorage[F, GlobalIncrementalSnapshot, GlobalSnapshotInfo],
-    singlePullLimit: Option[PosLong]
+    singlePullLimit: Option[PosLong],
+    maybeMajorityPeerIdSet: Option[NonEmptySet[PeerId]]
   ): GlobalL0Service[F] =
     new GlobalL0Service[F] {
 
       private val logger = Slf4jLogger.getLogger[F]
+      private val maybeMajorityPeerIds = maybeMajorityPeerIdSet.map(_.toNonEmptyList)
 
       def pullLatestSnapshot: F[(Hashed[GlobalIncrementalSnapshot], GlobalSnapshotInfo)] =
         globalL0ClusterStorage.getRandomPeer.flatMap { l0Peer =>
           l0GlobalSnapshotClient.getLatest(l0Peer).flatMap {
-            case ((snapshot, state)) =>
+            case (snapshot, state) =>
               snapshot.toHashedWithSignatureCheck.flatMap(_.liftTo[F]).map((_, state))
           }
         }
 
+      def pullLatestSnapshotFromMajority: F[(Hashed[GlobalIncrementalSnapshot], GlobalSnapshotInfo)] =
```

Comment:
I think we could have the names/functions other way around. Right now you are calling a function that states you will pull snapshot from majority but actually if the list of majority nodes is None you are not pulling from majority. Config decides whether we pull from majority or not so what I would do, would be to actually call a function with name `pullLatestSnapshot` and if the majority peers are set we will pull from majority if they are not we will simply pull from random peer. More general function based on configuration chooses the used logic.